src/ringbuffer_trait.rs

```diff
@@ -43,6 +43,12 @@ pub trait RingBufferWrite<T>: RingBuffer<T> {
     fn push(&mut self, value: T);
 }
 
+/// Similar to [`RingBufferWrite`] but only if [`T`] is [`Clone`].
+pub trait RingBufferWriteClone<T: Clone>: RingBuffer<T> {
```

Comment:
If I am not mistaken, this could be a part of RingBufferWrite as follows:

```rust
fn extend(&mut self, values: &[T]) where T: Clone;
```

I believe we already do this in other parts of the library.

Reply:
ah yes, good catch. I updated the PR